src/Microsoft.Diagnostics.NETCore.Client/DiagnosticsIpc/IpcEndpointConfig.cs

```diff
@@ -0,0 +1,156 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.Diagnostics.NETCore.Client
+{
+    internal class IpcEndpointConfig
+    {
+        public enum PortType
+        {
+            Connect,
+            Listen
+        }
+
+        public enum TransportType
+        {
+            NamedPipe,
+            UnixDomainSocket
```

Comment:
should this include TCP?

Reply:
I think it is better to add that once we have support for it.